### PR TITLE
⭐️ GCP asset inventory pack

### PR DESCRIPTION
**Pull request by scottford-io**
Screenshot 2023-01-09 at 4 57 47 PM

Signed-off-by: Scott Ford <scott@scottford.io>

### CODE REVIEW

File: core/mondoo-gcp-inventory.mql.yaml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    name: Mondoo mondoo-asset-inventory-gcp Asset Inventory
+    tags:
+      mondoo.com/platform: gcp,cloud
+      mondoo.com/category: best-practices
+    docs:
+      desc: |
+        The Mondoo GCP Inventory query pack gathers information about GCP projects for asset inventory.
+    filters:
+      - asset.platform == "gcp"
+      - uid: mondoo-asset-inventory-gcp-project-info
```
Comment (chris-rock) [inline core/mondoo-gcp-inventory.mql.yaml:12]: queries keyword is missing

File: core/mondoo-gcp-inventory.mql.yaml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+        docs: 
+          desc: |
+            This query gathers a list of all services enabled in the GCP Project
+        query: gcp.project.services.where( enabled == true ) { * }
+      - uid: mondoo-asset-inventory-gcp-gke-clusters-count
+        title: Gather count of GKE clusters in a GCP project
+        docs: 
+          desc: |
+            This query gathers a count of GKE clusters running in a GCP project
+        query: gcp.project.clusters.length
```
Comment (chris-rock) [inline core/mondoo-gcp-inventory.mql.yaml:47]: This was renamed to `gcp.project.gke.clusters`

**Review by chris-rock — CHANGES_REQUESTED** [review]
Great first version @scottford-io I added a few small things. LGTM once its done

File: core/mondoo-gcp-inventory.mql.yaml (hunk 1 above)
Comment (scottford-io) [reply core/mondoo-gcp-inventory.mql.yaml:12]: Wow...I am not sure how the pack is working at all. 🤔  Regardless, thanks finding that @chris-rock

File: core/mondoo-gcp-inventory.mql.yaml (hunk 2 above)
Comment (scottford-io) [reply core/mondoo-gcp-inventory.mql.yaml:47]: @chris-rock I just tested this with cnspec `7.11.0` and the query failed, and then upgraded to `7.12.0` and it worked. Are you ok with `gcp.project.gke.clusters` not being backwards compatible?

File: core/mondoo-gcp-inventory.mql.yaml [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        title: Gather data on GKE clusters
+        docs: 
+          desc: |
+            This query gathers all of the configuration data for GKE clusters within a project
+        query: gcp.project.clusters { * }
+      - uid: mondoo-asset-inventory-gcp-compute-instances-count
+        title: Gather a count of GCP compute instances in a GCP project
+        docs: 
+          desc: |
+            This query gathers a count of running GCP compute instances running in a GCP project
```
Comment (misterpantz) [inline core/mondoo-gcp-inventory.mql.yaml:58]:
```suggestion
            This query gathers a count of running GCP compute instances in a GCP project
```

**Review by misterpantz — COMMENTED** [review]
Awesome, @scottford-io!! Sorry I don't have time to edit this for consistency. I just reviewed it quickly and have one suggestion. 

AMAZING how quickly you did this. Thank you!

File: core/mondoo-gcp-inventory.mql.yaml (hunk 2 above)
Comment (chris-rock) [reply core/mondoo-gcp-inventory.mql.yaml:47]: Yes, we had `gcp.project.clusters` in evaluation and it turned out not to be the right place. The way to go is `gcp.project.gke.clusters`

**Review by chris-rock — APPROVED** [review]
Thank you @scottford-io for the great addition

**Pull request merged**